src/main/java/picard/illumina/parser/readers/CbclReader.java

```diff
@@ -269,7 +272,8 @@ private void advance() {
             for (int cycle = 0; cycle < outputLengths[read]; cycle++) {
                 CycleData currentCycleData = cycleData[totalCycleCount];
 
-                if (cachedTilePosition[totalCycleCount] >= cachedTile[totalCycleCount].length) {
+                if (cachedTilePosition[totalCycleCount] >= cachedTile[totalCycleCount].length
+                        || cachedTilePosition[totalCycleCount] >= cycleData[totalCycleCount].getTileInfo().getNumClustersInTile()) {
```

Comment:
I assume you are talking about USE_NEW_CONVERTER even though this comment is down here.Currently the old converter does not support cbcls. So yes you would have to set it to true.  I'm happy to have it detect this case (presence of cbcls), however once we can use the new converter with say HiSeqXs we might have to revisit this.